### DISCUSSION TITLE
Intellichlor Setting on Dashpanel/HA Integration Differs from ET Controller

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#861] [Q&A] [answered]
I have an ET2.4 and am running njs-PC 8.0.1 and Dashpanel 8.0.0.  I also run the HA Integration 0.4.12.  Everything has been running flawlessly for months.  Because the weather is cooler & my pool is in the shade most of the day, I have been turning down the Intellichlor setting using either/both the HA integration and the Dashpanel.  Both report the same setting and output level.  However, the chlorine level measured with a test kit has been increasing.

This morning I noticed the salt cell had 2 green LEDs for the cell output (40%), even though the setting in Dashpanel is 5%.  On the ET itself, the setting shows as 32%.  On the ET I lowered the setting to 20% and the salt cell switched to one green LED for the output, which I believe to be correct for 20%.  However,, Dashpanel & HA both continue to report a setting and output of 5%.

Everything else appears to be correct on the Dashpanel and HA integration: Pump Speed, Water Temperature, Heater Setting, Schedules for the filter pump and booster pump, Filter Pump Speed Setting, etc.

I've never noticed this inconsistency in behavior previously.  Can you kindly suggest how to troubleshoot, and hopefully correct, this inconsistency?

Thanks for your help!!!!

**Comment by rstrouse** [comment]
Actually once you get below 10% the lights switch to ~2% per light or 5%.

**Reply by FlaMike** [reply]
Thank you for the additional information regarding the salt cell LEDs.

I'm still flummoxed by the salt cell settings via njs-PC vs on the ET panel itself.  The salt concentration, which I believe is calculated via/by the salt cell is consistent whether I view it on the ET, via the njs-PC Dash Panel, or with the HA integration.  So, I'm guessing the challenge I have is not due to funky connections/wiring since both values are communicated via the RS-485 bus.

Any thoughts on how to diagnose/troubleshoot/correct this inconsistency is appreciated greatly!

**Reply by rstrouse** [reply]
Yes you should be able to see the exact communication from the cell in messageManager.  It will be what is being responded and what is being sent.

**Reply by FlaMike** [reply]
Just wanted to let you know I played around briefly with messageManager as suggested.  If I change the salt cell setpoint using dashPanel I see the messages going to the salt cell (with a notation for a "new" value) and from what little I can tell, it has the correct new value in the message field.  I also see acks going back & forth, but the cell setting does not appear to change.  

I'm going out of town for around 10 days, but when I return, I'll dig in deeper & will probably post screen shots & questions.  The skinny from what I can see is that messages are zipping along bidirectionally on the 485 bus between the salt cell & the ET.  Of  course, njs-PC sees them as well but the dashPanel and HA cell settings & output are not in synch with the "actual" cell settings, nor do they appear to be able to change it.  They also do not reflect changes made on the ET panel.  Sigh.

**Reply by FlaMike** [reply, marked as answer]
I no longer run the ET and am full Nixie now :-)

To close the loop, I believe the problem was the RS-485 surge suppressor I installed between the surge board and the ET & njsPC.  There's a little more information about my experiences & resolution of trying to get the Intellichlor to communicate with Nixie towards the end of this thread.

https://github.com/tagyoureit/nodejs-poolController/discussions/892
